### ISSUE TITLE
CSI-2 2 & 4 Lane Support

### ISSUE THREAD

**Issue by ETE-Design** [ayufan/camera-streamer#120]
Hi.

The Raspberry Pi 5 seems to Support CSI-2 - 4 Lane, and there is already some camera chip that support 4-lane. Wouldn't it give a performance gain to support it? And do you know how to make this part?

**Comment by kbingham** [comment]
Yes, the RPi5 supports 4 lane cameras. I have a couple here that I am testing. But this is probably a discussion more appropriate for the raspberry pi forums or kernel repository https://github.com/raspberrypi/linux/ rather than a camera streamer application.

**Comment by kbingham** [comment]
In terms of 'performance gain' ... it really depends on what you're trying to achieve. You can either transfer twice the pixels in the same amount of time, or you can use half the clock speeds to transfer the same quantity of data.

**Comment by ayufan** [comment]
@ETE-Design Unlikely to happen, since 4k60HZ are unlikely to be hardware encoded by the CPU. I will be testing this yet, but I would not have my hopes high.

**Comment by kbingham** [comment]
Indeed, you're going to hit encoder performance limitations long before you hit MIPI Data lane limitations I believe.

**Comment by kbingham** [comment]
@ayufan Note that Pi5 does not have a hardware encoder at all. RPi use ffmpeg in their rpicam-apps on pi5.

**Comment by ayufan** [comment]
@kbingham Yeah, I corrected my message. I know which makes even h264 1080p@60fps at the edge on pi5 :)

**Comment by kbingham** [comment]
https://github.com/raspberrypi/rpicam-apps/blob/main/encoder/libav_encoder.cpp

**Comment by ayufan** [comment]
I'm thinking the same, so I will use ffmpeg for that. There's no reason to reinvent a wheel.

**Comment by kbingham** [comment]
The awesome part then will mean your camera-streamer will really not be tied to RPi devices ... and it would run on /any/ libcamera supported platform.
